### PR TITLE
MINOR: use classic consumer with ZK mode for DeleteOffsetsConsumerGroupCommandIntegrationTest

### PR DESCRIPTION
**Pull request by FrankYang0529**
Ref: https://github.com/apache/kafka/pull/15766#discussion_r1585460093

Add ZK mode with classic consumer back to DeleteOffsetsConsumerGroupCommandIntegrationTest.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by chia7712** [comment]
Maybe we should merge `DeleteOffsetsConsumerGroupCommandIntegrationTest` and `DeleteConsumerGroupsTest`. They are used to test delete-related commands.

File: tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteOffsetsConsumerGroupCommandIntegrationTest.java [hunk 1]
```diff
@@ -88,7 +113,7 @@ public void testDeleteOffsetsNonExistingGroup() {
         }
     }
 
-    @ClusterTest
+    @ClusterTemplate("generator")
     public void testDeleteOffsetsOfStableConsumerGroupWithTopicPartition() {
         for (Map<String, Object> consumerConfig: consumerConfigs) {
```
Comment (chia7712) [inline tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteOffsetsConsumerGroupCommandIntegrationTest.java]: `consumerConfigs` need to be rewritten by `ClusterInstance#supportedGroupProtocols`

see https://github.com/apache/kafka/pull/15766/files#r1590222192

Comment (FrankYang0529) [reply tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteOffsetsConsumerGroupCommandIntegrationTest.java]: Updated it. Thanks.

File: tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteOffsetsConsumerGroupCommandIntegrationTest.java [hunk 2]
```diff
@@ -57,28 +56,47 @@
...
         this.clusterInstance = clusterInstance;
-        this.consumerConfigs = clusterInstance.isKRaftTest()
-                ? Arrays.asList(Collections.singletonMap(ConsumerConfig.GROUP_PROTOCOL_CONFIG, GroupProtocol.CLASSIC.name()),
-                Collections.singletonMap(ConsumerConfig.GROUP_PROTOCOL_CONFIG, GroupProtocol.CONSUMER.name()))
-                : Collections.singletonList(Collections.emptyMap());
     }
 
-    @ClusterTest
+    private static void generator(ClusterGenerator clusterGenerator) {
+        Map<String, String> serverProperties = new HashMap<>();
```
Comment (chia7712) [inline tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteOffsetsConsumerGroupCommandIntegrationTest.java]: Please use `ConsumerGroupCommandTestUtils#generator`

Comment (FrankYang0529) [reply tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteOffsetsConsumerGroupCommandIntegrationTest.java]: Updated it. Thanks.

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**